packages/engine/Source/Scene/SkyBox.js

```diff
@@ -141,10 +141,15 @@ SkyBox.prototype.update = function (frameState, useHdr) {
 
     if (typeof sources.positiveX === "string") {
       // Given urls for cube-map images.  Load them.
-      loadCubeMap(context, this._sources).then(function (cubeMap) {
-        that._cubeMap = that._cubeMap && that._cubeMap.destroy();
-        that._cubeMap = cubeMap;
-      });
+      loadCubeMap(context, this._sources).then(
+        function (cubeMap) {
+          that._cubeMap = that._cubeMap && that._cubeMap.destroy();
+          that._cubeMap = cubeMap;
+        },
+        function (error) {
+          console.error("Skybox cube map failed to load", error);
```

Comment:
Typically, we don't like to log to the console, and prefer throwing errors where possible so the user can deal with it as appropriate for their app. Though the fact that this is in the update loop complicates things.

1. Ideally, the async code would be removed from the update loop entirely, and there would be a top-level async function like `SkyBox.fromCubeMap()`  or `SkyBox.fromImages()`. But that may require more refactoring than you're willing to do at this time.
2. Next best would be to await the promise, catch the error, and throw it synchronously in the next call to `update`. That way the unit tests can use `pollToPromise` call `update` until an error is thrown or  `_cubeMap` is defined.
3. Finally, as a comprise, could we instead use a [`oneTimeWarning`](https://github.com/CesiumGS/cesium/blob/main/packages/engine/Source/Core/oneTimeWarning.js#L29)?

Reply:
2 is good.  That preserves the expectation that the error is getting thrown, but allows the test to avoid throwing the error (as `update` is never called  second time)